akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -364,4 +362,9 @@ object Source extends SourceApply {
     new Source(new ActorRefSource(bufferSize, overflowStrategy, DefaultAttributes.actorRefSource, shape("ActorRefSource")))
   }
 
+  def queue[T](bufferSize: Int, overflowStrategy: OverflowStrategy, timeout: FiniteDuration = 5.seconds): Source[T, T ⇒ Future[Boolean]] = {
```

Comment:
I know that the raw function type is elegant ("() ⇒ Future[Boolean]") but I would prefer a Trait instead with a named method, "offer()" for example.